webhook/workspace/handler/immutable.go

```diff
@@ -39,9 +39,17 @@ var ImmutableWorkspaceDiffOptions = []cmp.Option{
 }
 
 func (h *WebhookHandler) HandleImmutableMutate(_ context.Context, req admission.Request) admission.Response {
```

Comment:
If we take into account restricted annotation related changes, I think it makes sense to rework(at least rename) this HandleImmutable since it's not much anymore.

Maybe something like `HandleRestrictedAccessedObjOnUpdate` would suit a bit more now, but I'm not sure.